### ISSUE TITLE
508 Accessibility: Color contrast - (* Required)

### ISSUE THREAD

**Issue by grossb1** [department-of-veterans-affairs/va.gov-team#21848]
When reviewing the Project Cheetah workflow for color contrast issues I noticed that for a few of the colorblindness settings, the red font color for the `required` fields was a bit difficult to read.  This color is likely defined in a style-guide that is used more broadly across the app.

.../health-care/schedule-view-va-appointments/appointments/new-covid-19-vaccine-booking/received-dose
.../health-care/schedule-view-va-appointments/appointments/new-covid-19-vaccine-booking/facility
.../health-care/schedule-view-va-appointments/appointments/new-covid-19-vaccine-booking/clinic
.../health-care/schedule-view-va-appointments/appointments/new-covid-19-vaccine-booking/select-date-1
.../health-care/schedule-view-va-appointments/appointments/new-covid-19-vaccine-booking/contact-info

Steps to reproduce:
Navigate to one of the above pages
Activate your colorblind testing tool (I used the Colorblinding chrome plugin)
Sample the hex color of the `required` text (You can use something like Pixie to do this)
Sample the hex color of the background, it should be #FFFFFF
Go to a website like https://whocanuse.com/ and enter the hex values sampled above
Notice that the contrast test fails.

@outerpress

**Comment by outerpress** [comment]
@1Copenut it looks like the required red is failing contrast checks under some settings using the Colorblinding plugin.

I know this is a standard red/part of our color system - can you check/confirm these results?

**Comment by 1Copenut** [comment]
@outerpress & @grossb1 Excellent question. I had to look this one up myself.

I went to the WCAG 2.0 guidance for success criteria [1.4.3 - Contrast (Minimum)](https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-contrast.html). About halfway down the page was a paragraph that spoke directly to the situation of reduced contrast ratios for color vision deficiencies:

> Hues are perceived differently by users with color vision deficiencies (both congenital and acquired) resulting in different colors and relative luminance contrasts than for normally sighted users. Because of this, effective contrast and readability are different for this population. However, color deficiencies are so diverse that prescribing effective general use color pairs (for contrast) based on quantitative data is not feasible. Requiring good luminance contrast accommodates this by requiring contrast that is independent of color perception. Fortunately, most of the luminance contribution is from the mid and long wave receptors which largely overlap in their spectral responses. The result is that effective luminance contrast can generally be computed without regard to specific color deficiency, except for the use of predominantly long wavelength colors against darker colors (generally appearing black) for those who have protanopia. (We provide an advisory technique on avoiding red on black for that reason). For more information see [ARDITI-KNOBLAUCH] [ARDITI-KNOBLAUCH-1996] [ARDITI].

Reading through that a few times, I take this to mean our contrast ratios must line up to 4.5:1 (or greater) in general situations, but there is not a good way to account for color contrast ratios across the entire spectrum of color vision issues. This speaks to why we should never rely on color as our only means to communicate things like required fields, clickable links or buttons, or time-sensitive information.

tl;dr If the axe browser plugin is saying the color contrast ratios are valid, the Chrome dev tools are saying ratios are valid, and/or e2e tests are not failing, we are in the right place, and should remain vigilant for situations where we are relying on color to communicate intent.

**Comment by outerpress** [comment]
@1Copenut thanks for clarifying!

So looking at the [guidelines for testing with the Colorblindness plugin](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/staging-review-processes.md#colorblindness) it seems to say that you can use that tool for checking contrast ratios and other issues.

But from what you're saying it's mainly useful for testing/validating:
* All links, buttons, icons, and other elements retain their meaning when color blindness filters are enabled.
* Charts, maps, and tables should convey information without relying on color

Does that sound right?

**Comment by 1Copenut** [comment]
@outerpress Yes, our guidance is to use Colorblinding for validating items for intent and meaning when filters are enabled. I'll link to your comment on the other two issues asking about this item. Please feel free to close when you're ready.

**Comment by outerpress** [comment]
Thanks!
